File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/fragment/viewmodel/DataSettingViewModelFactory.java
package com.capstone.project_niyakneyak.ui.main.fragment.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class DataSettingViewModelFactory extends ViewModelProvider.AndroidViewModelFactory {
    private final Application application;
    public DataSettingViewModelFactory(Application application) {
        this.application = application;
    }
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(DataSettingViewModel.class)) {
            return (T) new DataSettingViewModel(application);
        } else {
            throw new IllegalArgumentException("Unknown ViewModel class");
        }
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/fragment/viewmodel/DataSettingViewModelFactory.kt
package com.capstone.project_niyakneyak.ui.main.fragment.viewmodel

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class DataSettingViewModelFactory(private val application: Application) :
    ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return if (modelClass.isAssignableFrom(DataSettingViewModel::class.java)) {
            DataSettingViewModel(application) as T
        } else {
            throw IllegalArgumentException("Unknown ViewModel class")
        }
    }
}